### ISSUE TITLE
tf.nn.elu returns NaN

### ISSUE THREAD

**Issue by aht** [tensorflow/tensorflow#5730]
I am using a fairly straight-forward FFN with ELU activations, and getting all NaN from the network output.

This comes in the 4th training epoch so the data did not have any NaN.

### Environment info
Operating System: Mac OS X 10.12.1

Installed version of CUDA and cuDNN: N/A. I'm using CPU version.

tensorflow version: 0.11.0rc2

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

    dense_output = tf.contrib.layers.stack(input, 
                                           tf.contrib.layers.fully_connected,
                                           [config.hidden_size, config.hidden_size // 2, config.hidden_size // 3, target_size],
                                           activation_fn=tf.nn.elu,
                                           weights_initializer=tf.contrib.layers.xavier_initializer(),
                                           scope='FFN')

### Logs or other output that would be helpful

https://gist.github.com/aht/5c719ce162158eaae9beb22555324f9d

**Comment by aht** [comment]
I am using MSE loss and the targets do not have NaN. This is evidenced by the fact that the training went thru 3 epochs without any issues.

Removing `activation_fn=tf.nn.elu` makes the problem goes away. Replacing with tf.nn.relu also works.

**Comment by aht** [comment]
I did find that my training data has some NaN. I don't know why it worked for 3 epochs. But we can close this for now.
